python/lsst/ts/imsim/obs_metadata.py

```diff
@@ -31,11 +34,87 @@ class ObsMetadata:
     ra: float  # ra in degrees
     dec: float  # dec in degrees
     band: str
-    zenith: float = 0.0
     rotator_angle: float = 0.0
     exp_time: float = 30.0
     mjd: float = 59580.0
     seq_num: int = 1
     raw_seeing: int = 0.5
     obs_id: str = """$f"IM_P_{astropy.time.Time(mjd, format='mjd').strftime('%Y%m%d')}_{seqnum:06d}" """
     focus_z: float = 0.0  # Defocal distance in mm
+    zenith: float = field(init=False)
+    parallactic_angle: float = field(init=False)
+
+    def __post_init__(self) -> None:
+        """Populate the zenith and parallactic angles
+        with values from init.
+        """
+        self.zenith = self.calc_zenith_angle()
+        self.parallactic_angle = self.calc_parallactic_angle()
+
+    def format_observation_info(self) -> (Observer, astropy.time, astropy.coordinates):
+        """
+        Get the observation info in the data structures needed
+        to calculate observer information such as parallactic angle
+        and zenith angle.
+
+        Parameters
+        ----------
+        obs_metadata : lsst.ts.imsim.ObsMetadata
+            ObsMetadata dataclass object with observation information.
+
+        Returns
+        -------
+        Observer
+            astroplan.Observer for observer located at Rubin Observatory
+        astropy.time
+            Observation time
+        astropy.coordinates
+            Observation boresight
+        """
+
+        # Observer located at Rubin
+        rubin_observer = Observer.at_site("cerro pachon")
+        time = astropy.time.Time(self.mjd, format="mjd")
+        boresight = astropy.coordinates.SkyCoord(
+            f"{self.ra}d", f"{self.dec}d", frame="icrs"
+        )
+
+        return rubin_observer, time, boresight
+
+    def calc_parallactic_angle(self) -> float:
+        """Calculate the parallactic angle so we know the
+        sky rotation angle on alt-az mount for the observation.
+
+        Parameters
+        ----------
+        obs_metadata : lsst.ts.imsim.ObsMetadata
+            ObsMetadata dataclass object with observation information.
+
+        Returns
+        -------
+        float
+            Parallactic Angle in degrees.
+        """
+
+        observer, time, boresight = self.format_observation_info()
+
+        return observer.parallactic_angle(time, boresight).deg
+
+    def calc_zenith_angle(self) -> float:
+        """Calculate the zenith angle so we can accurately
+        populate the imsim configuration.
+
+        Parameters
+        ----------
+        obs_metadata : lsst.ts.imsim.ObsMetadata
+            ObsMetadata dataclass object with observation information.
+
+        Returns
+        -------
+        float
+            Zenith Angle in degrees.
+        """
+
+        observer, time, boresight = self.format_observation_info()
+
+        return 90.0 - observer.altaz(time, boresight).alt.deg
```

Comment:
Maybe add a comment on the fact that altaz angle corresponds to the elevation angle, and therefore needs to be converted into zenith?